tooling/templatize/internal/end2end/e2e.go

```diff
@@ -87,6 +100,30 @@ func (e *e2eImpl) UseRandomRG() {
 	}
 	e.rgName = rg
 	e.SetConfig(config.Variables{"defaults": config.Variables{"rg": rg}})
+
+	return func() error {
+		subsriptionID, err := pipeline.LookupSubscriptionID(context.Background(), "ARO Hosted Control Planes (EA Subscription 1)")
+		if err != nil {
+			return err
+		}
+		cred, err := azidentity.NewDefaultAzureCredential(nil)
+		if err != nil {
+			return err
+		}
+		rgClient, err := armresources.NewResourceGroupsClient(subsriptionID, cred, nil)
+		if err != nil {
+			return err
+		}
+		rgDelResponse, err := rgClient.BeginDelete(context.Background(), e.rgName, nil)
+		if err != nil {
+			return err
+		}
+		_, err = rgDelResponse.PollUntilDone(context.Background(), nil)
```

Comment:
this could potentially take very long depending on the contents of the RG. i would not wait for successfull deletion here. not marking the RG with `persist:true` and relying on our regular cleanup mechanism should be sufficient.